### PR TITLE
#1129 Fixed VsReputation, added unit test

### PR DESCRIPTION
**Pull request by carlosmiranda**
#1129: Fixed `VsReputation`, users with higher reputation are now ranked higher. Also added (previously missing) unit test to verify correct behavior

### CODE REVIEW

**Comment by 0crat** [comment]
Job [#1319](https://github.com/zerocracy/farm/issues/1319) is now in scope, role is `REV`

**Comment by 0crat** [comment]
This pull request [#1319](https://github.com/zerocracy/farm/issues/1319) is assigned to @amihaiemil[/z](https://www.0crat.com/u/amihaiemil), here is [why](https://www.0crat.com/footprint/C3NDPUA8L/1840706465); the budget is 15 minutes, see [§4](http://www.zerocracy.com/policy.html#4); please, read [§27](http://www.zerocracy.com/policy.html#27) and when you decide to accept the changes, inform @g4s8[/z](https://www.0crat.com/u/g4s8) (the architect) right in this ticket; if you decide that this PR should not be accepted ever, also inform the architect; this [blog post](http://www.yegor256.com/2015/02/09/serious-code-reviewer.html) will help you understand what is expected from a code reviewer; there will be a monetary reward for this job

**Comment by carlosmiranda** [comment]
@amihaiemil ping

File: src/test/java/com/zerocracy/pm/staff/votes/VsReputationTest.java [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+ * @author Carlos Miranda (miranda.cma@gmail.com)
+ * @version $Id$
+ * @since 0.24
+ * @checkstyle JavadocMethodCheck (500 lines)
+ * @checkstyle MagicNumber (500 line)
+ */
+public final class VsReputationTest {
+    @Test
+    public void givesHigherVoteForHigherRep() throws Exception {
+        final String high = "high";
```
Comment (amihaiemil) [inline src/test/java/com/zerocracy/pm/staff/votes/VsReputationTest.java]: @carlosmiranda I suppose ``high`` and ``low`` are the Github users? Can you give them a better name? like ``amihaemil`` and ``carlosmiranda``?

File: src/main/java/com/zerocracy/pm/staff/votes/VsReputation.java [hunk 2]
```diff
@@ -70,16 +70,16 @@ public VsReputation(final Pmo pmo, final Collection<String> others) {
     public double take(final String login, final StringBuilder log)
```
Comment (amihaiemil) [inline src/main/java/com/zerocracy/pm/staff/votes/VsReputation.java]: @carlosmiranda As far as I understood, before, ``VsReputation`` was working in "negation": that is, it gave the lower reputation a higher score, but this score had a **low** or **negative** weight when used in ``Estimates``.

Now, the logic was reversed, so should we also edit the weight in ``Estimates``, to make it something positive?

File: src/main/java/com/zerocracy/pm/staff/votes/VsReputation.java [hunk 3]
```diff
@@ -70,16 +70,16 @@ public VsReputation(final Pmo pmo, final Collection<String> others) {
...
         final int mine = this.ranks.value().get(login);
-        final int smaller = new Filtered<>(
-            rank -> rank < mine,
+        final int larger = new Filtered<>(
+            rank -> rank > mine,
             this.ranks.value().values()
         ).size();
         log.append(
             Logger.format(
                 "Reputation %+d jobs is no.%d",
```
Comment (amihaiemil) [inline src/main/java/com/zerocracy/pm/staff/votes/VsReputation.java]: @carlosmiranda is `` %+d`` correct here, shouldn't it be ``%d``?

File: src/test/java/com/zerocracy/pm/staff/votes/VsReputationTest.java [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+import org.hamcrest.Matchers;
+import org.junit.Test;
+
+/**
+ * Test case for {@link VsReputation}.
+ * @author Carlos Miranda (miranda.cma@gmail.com)
+ * @version $Id$
+ * @since 0.24
+ * @checkstyle JavadocMethodCheck (500 lines)
+ * @checkstyle MagicNumber (500 line)
```
Comment (amihaiemil) [inline src/test/java/com/zerocracy/pm/staff/votes/VsReputationTest.java]: @carlosmiranda I think this is not needed anyomore

File: src/main/java/com/zerocracy/pm/staff/votes/VsReputation.java (hunk 2 above)
Comment (carlosmiranda) [reply src/main/java/com/zerocracy/pm/staff/votes/VsReputation.java]: @amihaiemil not sure what you mean, I don't see anything in `Estimates`. If you mean `Elections`, it is given a weight of 5 (positive) in `elect_performer.groovy`.

File: src/main/java/com/zerocracy/pm/staff/votes/VsReputation.java (hunk 3 above)
Comment (carlosmiranda) [reply src/main/java/com/zerocracy/pm/staff/votes/VsReputation.java]: @amihaiemil `%+d` will cause a sign to be displayed always (including a positive sign for positive values), which is helpful for logging reputation.

**Comment by carlosmiranda** [comment]
@amihaiemil changes pushed, also see comments. Thanks

File: src/main/java/com/zerocracy/pm/staff/votes/VsReputation.java (hunk 2 above)
Comment (amihaiemil) [reply src/main/java/com/zerocracy/pm/staff/votes/VsReputation.java]: @carlosmiranda Yes, you are right, I meant ``Elections``, not ``Estimates``, sorry.

**Comment by amihaiemil** [comment]
@carlosmiranda looks better, but now the CI is failing

**Comment by g4s8** [comment]
@carlosmiranda @amihaiemil CI error should be fixed in #1331, can we merge?

**Comment by g4s8** [comment]
@rultor merge

**Comment by 0crat** [comment]
@ypshenychka[/z](https://www.0crat.com/u/ypshenychka) please review this job completed by @amihaiemil[/z](https://www.0crat.com/u/amihaiemil), as in [§30](http://www.zerocracy.com/policy.html#30); the job will be fully closed and all payments will be made when the quality review is completed

**Comment by 0crat** [comment]
The job [#1319](https://github.com/zerocracy/farm/issues/1319) is now out of scope

**Comment by 0crat** [comment]
Payment to `ARC` for a closed pull request, as in [§28](http://www.zerocracy.com/policy.html#28): +10 point(s) just awarded to @g4s8[/z](https://www.0crat.com/u/g4s8)

**Comment by ypshenychka** [comment]
@amihaiemil According to our [QA Rules](http://www.zerocracy.com/policy.html#42):
>The code reviewer found at least three problems in the code.
Comments were mostly about design problems, not cosmetic issues.

Only two issues were found during code review. 
Please confirm that you'll try to find at least three major problems while future reviews.

**Comment by amihaiemil** [comment]
@ypshenychka I confirm. Generally, you see I always ask questioms but since PRs are usually small, it's not always possible to find 3 issues and I don't want to ask for variable name changes (when it is not really the case) just for the sake of it :D

**Comment by ypshenychka** [comment]
@amihaiemil nothing personal :) I understand that it's not always possible to find 3 issues, but you're trying - and that's great. But rules are strict. You know that :D

**Comment by ypshenychka** [comment]
@0crat quality acceptable

**Comment by amihaiemil** [comment]
@ypshenychka sure :D

**Comment by 0crat** [comment]
Order was finished, quality is "acceptable": +15 point(s) just awarded to @amihaiemil[/z](https://www.0crat.com/u/amihaiemil)

**Comment by ypshenychka** [comment]
@0crat status

**Comment by 0crat** [comment]
> @0crat status ([here](https://github.com/zerocracy/farm/issues/1319#issuecomment-402547605))

@ypshenychka This is what I know about this job in [C3NDPUA8L](https://www.0crat.com/p/C3NDPUA8L), as in [§32](http://www.zerocracy.com/policy.html#32):

  * The job [#1319](https://github.com/zerocracy/farm/issues/1319) is not in scope
  * Job [footprint](https://www.0crat.com/footprint/C3NDPUA8L?q=%7Bjob%3A%27gh%3Azerocracy%2Ffarm%231319%27%2Ctype%3A%7B%24not%3A%2F%5ENotify%2F%7D%7D) (restricted area)

**Comment by ypshenychka** [comment]
@g4s8 Could you please do a manual payment here? Looks like 0crat ignored me :)

**Comment by ypshenychka** [comment]
@g4s8 ping

**Pull request merged**